### PR TITLE
Added query_string LDAP config option

### PR DESCRIPTION
**Pull request by lsmith77**
docs for https://github.com/symfony/symfony/pull/21402

### CODE REVIEW

**Comment by nietonfir** [comment]
Maybe merge with #7414 ?

**Comment by lsmith77** [comment]
ah doh .. didn't see this PR :-/

lets wait for feedback.

for example I am not sure if it made sense adding another example or not like I did.

**Comment by nietonfir** [comment]
No problem. ;-) 
Your example looks fine to me.

**Comment by lsmith77** [comment]
integrated https://github.com/symfony/symfony-docs/issues/7414

**Comment by lsmith77** [comment]
the relevant code branch was merged

File: reference/configuration/security.rst [hunk 1]
```diff
@@ -446,6 +448,17 @@ placeholder will be replaced with the user-provided value (his login).
 Depending on your LDAP server's configuration, you may need to override
 this value.
 
+query_string
+............
+
+**type**: ``string`` **default**: ``null``
+
+This is the string which will be used to query for the DN. The ``{username}``
+placeholder will be replaced with the user-provided value (his login).
```
Comment (xabbuh) [inline reference/configuration/security.rst]: We should use "their" instead of "his".

File: reference/configuration/security.rst [hunk 2]
```diff
@@ -446,6 +448,17 @@ placeholder will be replaced with the user-provided value (his login).
...
+Depending on your LDAP server's configuration, you will need to override
+this value. This setting is only necessary if the users DN cannot be derived
```
Comment (xabbuh) [inline reference/configuration/security.rst]: user's

File: reference/configuration/security.rst [hunk 3]
```diff
@@ -446,6 +448,17 @@ placeholder will be replaced with the user-provided value (his login).
...
+statically using the `dn_string` config option.
```
Comment (xabbuh) [inline reference/configuration/security.rst]: we need double backticks here

File: security/ldap.rst [hunk 4]
```diff
@@ -382,5 +404,66 @@ Configuration example for HTTP Basic
...
+                            service="ldap"
+                            dn-string="dc=example,dc=com"
+                            query-string="(&amp;(uid={username})(memberOf=cn=users,ou=Services,dc=example,dc=com))" />
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        $container->loadFromExtension('security', array(
```
Comment (xabbuh) [inline security/ldap.rst]: we should add the `// app/config/security.php` comment above

File: security/ldap.rst [hunk 5]
```diff
@@ -382,5 +404,66 @@ Configuration example for HTTP Basic
...
+        # app/config/security.yml
+        security:
+            # ...
+
+            firewalls:
+                main:
+                    # ...
+                    form_login_ldap:
+                        login_path: login
+                        check_path: login_check
```
Comment (xabbuh) [inline security/ldap.rst]: Recently we updated the docs to use the same route for `login_path` and `check_path`. I think we should do that here too.

Comment (lsmith77) [inline security/ldap.rst]: if I look at form_login.rst it seems like those config options are simply omitted there .. that being said there are also some previously existing places in this file where we have the check_path in examples .. so should i remove login_path/check_path for all examples in this file?

Comment (lsmith77) [inline security/ldap.rst]:
```
diff --git a/security/ldap.rst b/security/ldap.rst
index bd812fd..d91e977 100644
--- a/security/ldap.rst
+++ b/security/ldap.rst
@@ -310,8 +310,6 @@ Configuration example for form login
                 main:
                     # ...
                     form_login_ldap:
-                        login_path: login
-                        check_path: login_check
                         # ...
                         service: ldap
                         dn_string: 'uid={username},dc=example,dc=com'
@@ -329,8 +327,6 @@ Configuration example for form login
             <config>
                 <firewall name="main">
                     <form-login-ldap
-                            login-path="login"
-                            check-path="login_check"
                             service="ldap"
                             dn-string="uid={username},dc=example,dc=com" />
                 </firewall>
@@ -343,8 +339,6 @@ Configuration example for form login
             'firewalls' => array(
                 'main' => array(
                     'form_login_ldap' => array(
-                        'login_path' => 'login',
-                        'check_path' => 'login_check',
                         'service' => 'ldap',
                         'dn_string' => 'uid={username},dc=example,dc=com',
                         // ...
@@ -419,8 +413,6 @@ Configuration example for form login and query_string
                 main:
                     # ...
                     form_login_ldap:
-                        login_path: login
-                        check_path: login_check
                         # ...
                         service: ldap
                         dn_string: 'dc=example,dc=com'
@@ -439,8 +431,6 @@ Configuration example for form login and query_string
             <config>
                 <firewall name="main">
                     <form-login-ldap
-                            login-path="login"
-                            check-path="login_check"
                             service="ldap"
                             dn-string="dc=example,dc=com"
                             query-string="(&amp;(uid={username})(memberOf=cn=users,ou=Services,dc=example,dc=com))" />
@@ -455,8 +445,6 @@ Configuration example for form login and query_string
             'firewalls' => array(
                 'main' => array(
                     'form_login_ldap' => array(
-                        'login_path' => 'login',
-                        'check_path' => 'login_check',
                         'service' => 'ldap',
                         'dn_string' => 'dc=example,dc=com',
                         'query_string' => '(&(uid={username})(memberOf=cn=users,ou=Services,dc=example,dc=com))',
```

Comment (xabbuh) [inline security/ldap.rst]: looks good to me too

**Comment by lsmith77** [comment]
@xabbuh ok .. all comments should be addressed

**Comment by xabbuh** [comment]
👍 

Status: Reviewed

**Comment by xabbuh** [comment]
Thank you @lsmith77.

**Comment by xabbuh** [comment]
And thank you very much @nietonfir!

**Pull request merged**